src/onnx/parse_matmul.cpp

```diff
@@ -200,23 +368,50 @@ struct parse_matmul : op_parser<parse_matmul>
             auto s0_lens        = a0->get_shape().lens();
             auto s1_lens        = a1->get_shape().lens();
 
-            if(not is_quant_dot and args.size() > 2)
+            if(not is_quant_dot and args.size() > 2 and not has_scales)
```

Comment:
Would it be simpler to just check if it is just a `dot`, instead of looking for `quant_dot` and `quant_dot_scaled`, as this clause seems to be doing? Thanks.

Reply:
Sure that can easily be swapped

Reply:
Done